packages/xo-server/src/xapi/index.js

```diff
@@ -1122,14 +1122,32 @@ export default class Xapi extends XapiBase {
       return defaultSr.$ref
     })
 
+    const hostPbds = new Set(host.PBDs)
+    const connectedSrs = new Map()
+    const isConnectedSr = sr => {
+      let isConnected = connectedSrs.get(sr.$ref)
+      if (isConnected === undefined) {
+        isConnected = sr.PBDs.find(ref => hostPbds.has(ref))
+        connectedSrs.set(sr.$ref, isConnected)
+      }
+      return isConnected !== undefined
+    }
+
     // VDIs/SRs mapping
+    // - If SR was explicitly passed: use it
+    // - Else if VDI SR is reachable from the destination host: use it
+    // - Else: use the default SR for migration or default SR for pool
```

Comment:
VDI:
- map's SR
- main SR
- default SR
- error

VDI-snapshot:
- if active VDI exists: `undefined` (will be migrated to same SR as active VDI)
- main SR
- default SR
- error